pyproject.toml

```diff
@@ -180,6 +165,51 @@ norecursedirs = [".git", ".tox", ".env", "dist", "build"]
 python_files = ["test_*.py", "*_test.py", "tests.py"]
 
 
+[tool.ruff]
+exclude = [
+  ".git",
+  "__pycache__",
+  "src/vendor",
+  "build",
+  "dist",
+  ".venv",
+  "venv",
+]
+# ignore = [
+#     "E501",
+#     "F401",
+#     "F403",
+# ]
+extend-select = [
+  "B",
+  "C90",
+  "D",
+  "I",
+  "Q",
+  "UP",
+  "W"
+]
+ignore = [
+  "E501",
+]
+line-length = 80
+
+src = [
```

Comment:
does this work as we get these in pre-commit.ci

```console
scripts/download_weights.py:1:1: D100 Missing docstring in public module
scripts/download_weights.py:6:7: D101 Missing docstring in public class
scripts/download_weights.py:23:5: D103 Missing docstring in public function
```

Reply:
https://github.com/charliermarsh/ruff/tree/main#src only defines what is considered internal stuff if I understand the options.

Pushed a fix.

Reply:
ok, but now `scripts` is ignored by all checks. previously it was only ignored by `pydocstyle`, is there a way to ignore on tool level?

Reply:
```toml
 [tool.ruff.per-file-ignores]
"scripts/download_weights.py" = ["D100", "D101", "D103"]
```

Should work.